### DISCUSSION TITLE
Searching for files in 7-zip archives?

### DISCUSSION THREAD

**Discussion by szpak** [doublecmd/doublecmd#837] [General]
It is possible to search in ZIP archives, however, for 7-zip archives, it is "not supported". I wonder, if it is an underlying plugin/implementation limitation or it could work.

In fact, I'm not sure how 7-zip archives are handled in a vanilla Double Command 1.0.10 on Linux. In the plugins, I see a bundled `zip.wcx` (with "Allow searching for text in archives"). However, the `.7z` extension is not on the list and I don't see any other plugin with `.7z` support. Is it implemented in the "core"?

**Comment by alexx2000** [comment]
7-Zip attached as [external archiver](https://doublecmd.github.io/doc/en/multiarc.html#archivers). Search does not work in this case. You can try to install [P7Z Usr](https://github.com/doublecmd/doublecmd/wiki/Plugins#user-content-P7Z_Usr) or [libarchive](https://github.com/j2969719/doublecmd-plugins/tree/master/plugins/wcx/libarchive_crap) plugins. You can download binary versions from [here](https://disk.yandex.com/d/jhgHgkt1ypuhPw/other).

**Reply by szpak** [reply]
Thanks for your explanations. They don't look like actively maintained, so I will probably stay with the external archiver.

**Comment by Skif-off** [comment]
> so I will probably stay with the external archiver.

p7zip? Then it sounds ironic :)

Both plugins just work.

The libarchive plugin uses [multi-format archive and compression library](https://www.libarchive.org/) and most likely your distribution contains the latest version of this library.

P7Z Usr uses `7z.so ` from the system (if compiled with p7zip) or you can use this plugin with the current version of 7-Zip (the source code of P7Z Usr with minor changes is in the cloud).

**Reply by szpak** [reply]
Thanks for your comment.

> The libarchive plugin uses [multi-format archive and compression library](https://www.libarchive.org/) and most likely your distribution contains the latest version of this library.

My option was based on the fact that while the lib itself is actively maintained, but the plugin DCmd plugin is [marked](https://github.com/j2969719/doublecmd-plugins/tree/master/plugins/wcx/libarchive_crap) as "not actively maintained":

```
doublecmd-plugins/plugins/wcx/libarchive_crap/

Notes
    abandoned
    potentially dangerous
    some limitations
```

Similarly, the latest version of "P7Z Usr" is [dated to 2016](https://doublecmd.h1n.ru/viewtopic.php?f=5&t=3339).

Nevertheless, it's good they are still functional.

**Reply by j2969719** [reply]
more detailed explanation on libarchive_crap:
initially, was hoping to make a simple plugin, but in the end it turned out to be stupid shit code as usual. ideally, it would need to be rewritten, but im too lazy and not sure that will get something more adequate

not entirely sure whether it is possible to modify existing archives with libarchive, so every time something is added/removed to the existing archive, a new archive is created **using tempnam**, which eventually replaces the old one

if the old archive was encrypted or, say, with ultracompression, then when the archive is modified, it will eventually be replaced by a simple archive without encryption. also, libarchive itself, if not mistaken, had some limitations in terms of supporting solid rar or encryption for everything except zip

personally use it for tar.* and mtree but would definitely not advise other people to use this crap
